cli/src/cli_util.rs

```diff
@@ -2600,3 +2602,37 @@ impl CliRunner {
         exit_code
     }
 }
+
+fn map_clap_cli_error(
+    cmd_err: CommandError,
+    ui: &Ui,
+    layered_configs: &LayeredConfigs,
+) -> CommandError {
+    if let CommandError::ClapCliError { ref err, hint: _ } = cmd_err {
```

Comment:
nit: we can also use `let CommandError::ClapCliError { err, hint: _ } = &cmd_err else { return cmd_err; };` pattern (if you prefer.)

And maybe better to require `hint: None`.